keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -531,6 +619,66 @@ Below is the proposed strategy for doing so:
+
+| field               | type   | process isolated field                      | hyperv filed         | notes                                                                                                                                                               |
+| ------------------- | ------ | ------------------------------------------- | -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| timestamp           | int64  | ✅                                           | ✅                    |                                                                                                                                                                     |
+| working\_set\_bytes | uint64 | memory\_usage\_private\_working\_set\_bytes | working\_set\_bytes? | // The amount of working set memory in bytes.<br><br>Is hyper-v working\_set\_bytes same as private\_working\_set\_bytes                                            |
+| available\_bytes    | uint64 |                                             | available\_memory    |   We should be able to return this. It is the limit set on the job object.  https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-jobobject_extended_limit_information <br><br>  // Available memory for use. This is defined as the memory limit - workingSetBytes.<br><br> |
+| usage\_bytes        | uint65 | ?                                           | ?                    | // Total memory in use. This includes all memory regardless of when it was accessed. Is this cumultive memory usage?   |
+| rss\_bytes          | uint66 | n/a                                         | n/a                  | windows doesn't have rss. Cannot report rss                                                                                                                                              |
+| page\_faults        | uint67 |                                             |                      | not reported. It may be possible use `TotalPageFaultCount` from https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-jobobject_basic_accounting_information                            |
+| major\_page\_faults | uint68 |                                             |                      | not reported. Windows does not make a distinction here                                                                                                          |
+
+**Process isolated** also has
+
+- memory_usage_commit_bytes
+- memory_usage_commit_peak_bytes 
+
+**Hyperv** also has 
+
+- virtual_node_count
+- available_memory_buffer
+- reserved_memory 
+- assigned_memory 
+- slp_active 
+- balancing_enabled
+- dm_operation_in_progress 
+
+These are not used currently but are be very specific to each implementation and could be collected by specialized tools on a as needed basis.
```

Comment:
or maybe reported as an unstructured Metric through the endpoint formally known as /metrics/cadvisor?

Reply:
had the same thoughts, exposing them through that would be a good way to provide them to end users. If kubelet ends up needing them for decisions, we could introduce them to the formal Pod Stats call but can be done at a later time if need arises